### PR TITLE
Add Guidelines for Platforms Tested in CI

### PR DESCRIPTION
**Pull request by MichaelSimons**
This is an attempt at fixing https://github.com/dotnet/source-build/issues/2813.  

Note: this aligns with the current CI legs with the addition of a Debian leg.  This new leg feels more tenable now that https://github.com/dotnet/source-build/issues/3463 is fixed. 

Once I get some initial feedback, I will broaden the reviewers to include the distro maintainer alias.

### CODE REVIEW

File: Documentation/ci-platform-coverage-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+1. Only test officially supported distros. Community supported distros will not be covered.
+
+**Note:** There will be conflicts within these guidelines.  When they occur, it is an indication
+that multiple legs will be needed. For example, Ubuntu is based on Debian. However, .NET is included
+in Ubuntu's repositories by default but not Debian's (at the time of writing). As such, it is best
+to include legs for both Ubuntu and Debian.
+
+## Distro Versions
+
+1. It is not feasible to cover all supported distro versions. Judiciously select the versions to test.
```
Comment (mthalman) [inline Documentation/ci-platform-coverage-guidelines.md:22]: This needs more clarity. What guidelines will be used to determine which versions are selected?

File: Documentation/ci-platform-coverage-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+the source build CI.
+
+## Distro Families
+
+1. Prefer testing base distros of families over derivatives.
+1. Prioritize testing distros that source build .NET.
+1. Use CentOS Stream instead of Red Hat because it is the free alternative.
+1. Include a permutation of distros for the supported C standard library implementations
+(e.g. glibc and musl).
+1. Only test officially supported distros. Community supported distros will not be covered.
```
Comment (mthalman) [inline Documentation/ci-platform-coverage-guidelines.md:13]: When referring to "supported", it needs to be clarified what that's referring to. I'm assuming it means "supported by .NET". The same thing applies to the Architectures section.

File: Documentation/ci-platform-coverage-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Architectures
+
+1. All officially supported architectures (e.g. amd64 and arm64) will be covered. Community supported
+architectures will not be covered.
+
+## Permutations
+
+It is not the intent to test every permutation of distro family, distro version, and architecture.
+Rather, smart decisions should be made to utilize resources. For example, test the oldest LTS version
+of debian on amd64 and the latest LTS version of ubuntu on arm64.
```
Comment (mthalman) [inline Documentation/ci-platform-coverage-guidelines.md:33]:
```suggestion
of Debian on amd64 and the latest LTS version of Ubuntu on arm64.
```

File: Documentation/ci-platform-coverage-guidelines.md (hunk 2 above)
Comment (MichaelSimons) [reply Documentation/ci-platform-coverage-guidelines.md]: Reworded this to be 

```
Only test [distros that are officially supported by .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md#net-supported-os-policy).
Community supported distros will not be covered.
```

I refactored the architecture section to list amd64 and arm64 explicitly.  I decided this after realizing arm32 is officially supported by .NET but not something we support for source-build.

File: Documentation/ci-platform-coverage-guidelines.md (hunk 1 above)
Comment (MichaelSimons) [reply Documentation/ci-platform-coverage-guidelines.md]: Added

File: Documentation/package-dependency-flow.md [hunk 4]
```diff
@@ -42,8 +42,8 @@ ability to service the product.
 
 ## Package Versions
 
-Package dependencies that defined using 
-[Arcade's Darc patterns](https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md)
+Package dependencies defined using 
+[Arcade's dependency patterns](https://github.com/dotnet/arcade/blob/main/Documentation/Darc.md)
```
Comment (MichaelSimons) [inline Documentation/package-dependency-flow.md:46]: I missed committing this change in my last doc update so I am including it here.

File: Documentation/ci-platform-coverage-guidelines.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Community supported distros will not be covered.
+
+**Note:** There will be conflicts within these guidelines.  When they occur, it is an indication
+that multiple legs will be needed. For example, Ubuntu is based on Debian. However, .NET is included
+in Ubuntu's repositories by default but not Debian's (at the time of writing). As such, it is best
+to include legs for both Ubuntu and Debian.
+
+## Distro Versions
+
+When selecting which distro versions to test the .NET version in development, it is important to take into account which distro versions will be insupport at the time of the .NET release.  Don't worry about testing .NET on distro versions that will be or nearing EOL on the .NET release day.
```
Comment (mthalman) [inline Documentation/ci-platform-coverage-guidelines.md:23]:
```suggestion
When selecting which distro versions to test the .NET version in development, it is important to take into account which distro versions will be in support at the time of the .NET release.  Don't worry about testing .NET on distro versions that will be or nearing EOL on the .NET release day.
```

**Review by mthalman — APPROVED** [review]

**Pull request merged**